subsys/shell/shell_uart.c

```diff
@@ -111,6 +111,19 @@ static void uart_tx_handle(const struct device *dev,
 	len = ring_buf_get_claim(sh_uart->tx_ringbuf, (uint8_t **)&data,
 				 sh_uart->tx_ringbuf->size);
 	if (len) {
+#ifdef CONFIG_UART_LINE_CTRL
+		/* Wait for DTR signal before sending anything to output. */
+		uint32_t dtr = 0U;
+
+		while (true) {
+			uart_line_ctrl_get(sh_uart->ctrl_blk->dev, UART_LINE_CTRL_DTR, &dtr);
```

Comment:
you should check return code and break if `-ENOSYS` or `-ENOTSUP` is returned. You can remove `ifdef` then since if it is disabled then at compile time `uart_line_ctrl_get` will resolve to `-ENOTSUP` and code will be removed.

Reply:
Right, or how about just checking 0 value? I guess we don't want to go further in other cases than success.

Reply:
it is possible that line control is enabled but not on all instances.

Additionally, check for `CONFIG_UART_LINE_CTRL` might make sense for userspace since `uart_line_ctrl_get` is a syscall so it will probably not resolve to fixed value. Maybe additional break like
```
`if if(!IS_ENABLED(CONFIG_UART_LINE_CTRL)) break;